examples/module1/lesson1/solver/App.tsx

```diff
@@ -1,58 +1,30 @@
 import React, { useState } from 'react';
-import { f1, f2, f3, f4 } from './functions';
+import InputComponent from './component/InputComponent';
+import ButtonComponent from './component/ButtonComponent';
+import { add, divide, multiply, subtract } from './functions';
 
 const App = () => {
-  const [numA, setNumA] = useState<number>(0);
-  const [numB, setNumB] = useState<number>(0);
-  const [numC, setNumC] = useState<number | string>(0);
+  const [firstNumber, setFirstNumber] = useState<number>(0);
+  const [secondNumber, setSecondNumber] = useState<number>(0);
+  const [result, setResult] = useState<number | string>(0);
 
-  const doWork = (func: (a: number, b: number) => number) => {
-    setNumC(func(numA, numB));
+  const calculate = (func: (a: number, b: number) => number) => {
+    setResult(func(firstNumber, secondNumber));
   };
 
   return (
     <div>
       <div className="grid grid-cols-2 gap-x-4">
-        <input
-          type="number"
-          className="rounded-md shadow-md p-4"
-          value={numA}
-          onChange={(e) => setNumA(parseFloat(e.target.value))}
-        />
-        <input
-          type="number"
-          className="rounded-md shadow-md p-4"
-          value={numB}
-          onChange={(e) => setNumB(parseFloat(e.target.value))}
-        />
+        <InputComponent value={firstNumber} onChange={(e: any) => setFirstNumber(parseFloat(e.target.value))}/>
```

Comment:
za to 'any' powinno być po łapach linijką 😄 . Nie rozwodząc się zbyt długo, pewnie jesteś świadoma że danie any to jak wyłączenie typescripta, więc pytanie po co używać narzędzia któremu potem wyłącza się jego go główną funkcjonalność. Jeśli powodem była niewiedza to dam podpowiedzieć, ChangeEvent<HTMLInputElement>.